app/lib/paypal_client.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+module PaypalClient
+  module_function
+
+  require 'net/http'
+
+  STATUS = {
+    approved:  'APPROVED',
+    active:    'ACTIVE',
+    pending:   'APPROVAL_PENDING',
+    suspended: 'SUSPENDED',
+    expired:   'EXPIRED',
+    cancelled: 'CANCELLED'
+  }.freeze
+
+  PAYPAL_DOMAIN = ENV['PAYPAL_DOMAIN']
+
+  def get_subscription reference_id
+    result = request_get '/v1/billing/subscriptions/' + reference_id
+
+    parse_subscription(result) || { status: :no_results }
+  end
+
+  def parse_subscription result
+    case result['status']
+    when STATUS[:approved], STATUS[:active]
+      {
+        status:      :paid,
+        approved_at: Time.parse(result['billing_info']['last_payment']['time']),
+        paid_until:  Time.parse(result['billing_info']['next_billing_time'])
+      }
+    when STATUS[:pending]
+      {
+        status:     :in_process
```

Comment:
Con esta indentación creo que te pasaste... te estás haciendo adicto =P
```suggestion
        status: :in_process
```

Reply:
Jajaja pase "la vista" mil veces y nunca me molesto loco xD